### PR TITLE
Created update_csv_values() for randomizing dataset

### PR DESCRIPTION
**Pull request by rplevka**
- created `modify_csv_values` function for replacing the key values with keeping the entity mapping of the original data set. This allows us to implement the tests for `hammer import` feature with generated strings the same way as other features.
 ```bash
$ nosetests test_import.py
.....
----------------------------------------------------------------------
Ran 5 tests in 194.780s

OK
```
- swapped two tests to keep organization-related tests grouped.
- fixed the cleanup (added removal of the locally-generated files)

### CODE REVIEW

File: tests/foreman/cli/test_import.py [hunk 1]
```diff
@@ -44,8 +45,8 @@ def csv_to_dataset(csv_file):
 
 
 def build_csv_file(rows=None):
-    """Generates a csv file, feeds it by the provided data
-    (a list of dictionary objects) and returns a path to it
+    """Generate a csv file, feed it by the provided data
+    (a list of dictionary objects) and return a path to it
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_import.py]: :+1: For using present-tense language.

File: tests/foreman/cli/test_import.py [hunk 2]
```diff
@@ -59,6 +60,62 @@ def build_csv_file(rows=None):
...
+
+    :param csv_file: A string. The path to a CSV file that resides
+    on a remote server.
+
+    :param new_data: A list of dictionary objects. Each object has to
+    contain an id corresponding to organization_id of the record being
+    modified and the key-value pairs of attributes to be changed for
+    the particular records
+
+    :returns: A string. The path to a locally created CSV file
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_import.py]: Make sure to indent every line of text that belongs to a Sphinx directive:

```
    :param csv_file: A string. The path to a CSV file that resides on a remote
        server.
    :param new_data: A list of dictionary objects. Each object has to contain
        an id corresponding to organization_id of the record being modified and
        the key-value pairs of attributes to be changed for the particular
        records
    :returns: A string. The path to a locally created CSV file
```

Comment (Ichimonji10) [inline tests/foreman/cli/test_import.py]: Can you provide a brief example of what form the `new_data` argument should take?

    {???}

File: tests/foreman/cli/test_import.py [hunk 3]
```diff
@@ -59,6 +60,62 @@ def build_csv_file(rows=None):
...
+    # if new_data is not specified, no change happens
+    if new_data is None:
+        return csv_file
+
+    result = csv_to_dataset(csv_file)
+    for change in new_data:
+        for record in result:
+            if record['organization_id'] == change['org_id']:
+                for key in change.keys():
+                    record[key] = change[key]
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_import.py]: These two lines can be replaced with:

```python
record.update(change)
```

File: tests/foreman/cli/test_import.py [hunk 4]
```diff
@@ -59,6 +60,62 @@ def build_csv_file(rows=None):
...
+
+def positive_create_org_data():
+    return ([
+        {u'org_id': u'1', u'organization': gen_string('alphanumeric')},
+        {u'org_id': u'2', u'organization': gen_string('alphanumeric')},
+        {u'org_id': u'3', u'organization': gen_string('alphanumeric')},
+    ],)
+
+
+def positive_create_usr_data():
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_import.py]: Can this be renamed to `positive_create_user_data`?

File: tests/foreman/cli/test_import.py [hunk 5]
```diff
@@ -78,7 +135,13 @@ def test_import_orgs_default(self):
         @assert: 3 Organizations are created
 
         """
-        files = prepare_import_data()[1]
+        tmp_dir, files = prepare_import_data()
+        new_dataset = modify_csv_values(
+            files['users'], *positive_create_org_data()
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_import.py]: Nitpick: Place each argument on its own line.

File: tests/foreman/cli/test_import.py [hunk 6]
```diff
@@ -132,12 +195,44 @@ def test_import_orgs_manifests(self):
...
+        )
+        ssh.upload_file(new_dataset, new_dataset)
+        files['users'] = new_dataset
+        self.assertEqual(
+            Import.organization({'csv-file': files['users']}).return_code, 0)
+        self.assertEqual(
+            Import.organization({
+                'csv-file': files['users']
+            }).stdout, [u'Summary', u'  No action taken.', u'']
+        )
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_import.py]: How about:

```python
        self.assertEqual(
            Import.organization({'csv-file': files['users']}).stdout,
            [u'Summary', u'  No action taken.', u'']
        )
```

**Comment by Ichimonji10** [comment]
ACK pending comments

File: tests/foreman/cli/test_import.py (hunk 3 above)
Comment (rplevka) [inline tests/foreman/cli/test_import.py]: :+1: for this...i knew such method must exist!

**Comment by elyezer** [comment]
ACK, will leave to @Ichimonji10 take a second look.

**Comment by Ichimonji10** [comment]
`make docs` produces several warnings. This change fixes them:

```diff
diff --git a/tests/foreman/cli/test_import.py b/tests/foreman/cli/test_import.py
index 32c1102..0c3a336 100644
--- a/tests/foreman/cli/test_import.py
+++ b/tests/foreman/cli/test_import.py
@@ -24,7 +24,7 @@ def csv_to_dataset(csv_file):
     Read a remote CSV file, process it and return it.
 
     :param csv_file: A string. The path to a CSV file that resides
-    on a remote server.
+        on a remote server.
 
     :returns: A dictionary holding the contents of the CSV file.
 
@@ -70,20 +70,21 @@ def update_csv_values(csv_file, new_data=None):
     :param new_data: A list of dictionary objects. Each object has to
         contain an org_id key corresponding to organization_id of the record
         being updated and the key-value pairs of attributes to be changed for
-        the particular records.
-        E.g.:
-        [
-            {
-                u'org_id': u'1',
-                u'organization': u'updated_organization_name_1',
-                u'username': u'updated_user_name_1',
-            },
-            {
-                u'org_id': u'2',
-                u'organization': u'updated_organization_name_2',
-                u'username': u'updated_user_name_2',
-            }
-        ]
+        the particular records. For example::
+
+            [
+                {
+                    u'org_id': u'1',
+                    u'organization': u'updated_organization_name_1',
+                    u'username': u'updated_user_name_1',
+                },
+                {
+                    u'org_id': u'2',
+                    u'organization': u'updated_organization_name_2',
+                    u'username': u'updated_user_name_2',
+                }
+            ]
+
     :returns: A string. The path to a locally created CSV file
 
     """
```

**Comment by Ichimonji10** [comment]
APT

**Pull request merged**